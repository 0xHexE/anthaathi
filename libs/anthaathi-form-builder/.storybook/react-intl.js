const locales = ['en-US'];

const messages = locales.reduce(
  (acc, lang) => ({
    ...acc,
    [lang]: import(`../locales/compiled-locales/${lang}.json`), // whatever the relative path to your messages json is
  }),
  {},
);

const formats = {};

export const reactIntl = {
  defaultLocale: 'en-US',
  locales,
  messages,
  formats,
};
