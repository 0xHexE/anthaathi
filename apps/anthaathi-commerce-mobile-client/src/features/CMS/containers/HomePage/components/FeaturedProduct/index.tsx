import * as React from 'react';
import {useIntl} from 'react-intl';
import {Image, Pressable, View} from 'react-native';
import {Badge, Button, Text} from 'react-native-paper';
import {useResponsiveValue} from '../../../../utils/useResponsiveValue';

export interface ProductDetails {
  name: string;
  description: string;
  price: string;
  image: string[];
}

export interface ProductDetailsProps {
  productInfo: ProductDetails;
  handleAddToCart?: () => {};
  handleBuyItNow?: () => {};
}

function FeaturedProduct(props: ProductDetailsProps) {
  const itemWidth = useResponsiveValue(['80%', '70%', '70%', '70%']);
  const [imageIndex, setImageIndex] = React.useState(0);
  const intl = useIntl();

  return (
    <View
      style={{
        marginVertical: 10,
        marginHorizontal: 12,
        flexDirection: itemWidth === '80%' ? 'column' : 'row',
      }}
    >
      {itemWidth === '80%' && (
        <Text
          variant="titleLarge"
          style={{marginBottom: 9, fontSize: 20, fontWeight: '600'}}
        >
          {props.productInfo.name}
        </Text>
      )}
      <View
        style={[
          {
            alignItems: 'center',
            justifyContent: 'center',
          },
          itemWidth !== '80%' && {width: '50%'},
        ]}
      >
        <Image
          style={{
            height: 280,
            width: itemWidth,
            borderRadius: 4,
          }}
          source={{
            uri: props.productInfo.image[imageIndex],
          }}
        />

        <ImageSelection
          id={imageIndex}
          onChange={setImageIndex}
          imageLength={props.productInfo.image.length}
        />
      </View>
      <View style={[itemWidth !== '80%' && {width: '50%'}]}>
        {itemWidth !== '80%' && (
          <Text
            variant="titleLarge"
            style={{marginBottom: 9, fontSize: 20, fontWeight: '600'}}
          >
            {props.productInfo.name}
          </Text>
        )}

        <Pricing price={props.productInfo.price} />
        <Text
          variant="titleLarge"
          style={{marginBottom: 9, fontSize: 12, fontWeight: '400'}}
        >
          {props.productInfo.description}
        </Text>
        <Button
          mode="outlined"
          onPress={props.handleAddToCart}
          labelStyle={{color: '#313652'}}
          style={{borderColor: '#313652', borderRadius: 1, marginBottom: 10}}
        >
          {intl.formatMessage({defaultMessage: 'Add to cart'})}
        </Button>

        <Button
          mode="contained"
          onPress={props.handleBuyItNow}
          labelStyle={{color: '#ffffff'}}
          style={{
            backgroundColor: '#313652',
            borderRadius: 1,
            marginBottom: 10,
          }}
        >
          {intl.formatMessage({defaultMessage: 'Buy it now'})}
        </Button>
      </View>
    </View>
  );
}

const Pricing = ({price}: {price: string}) => {
  const intl = useIntl();
  return (
    <>
      <Text variant="titleLarge" style={{fontSize: 16, fontWeight: '500'}}>
        {intl.formatMessage({defaultMessage: 'Price'})}
      </Text>
      <Text
        variant="titleLarge"
        style={{marginBottom: 9, fontSize: 16, fontWeight: '500'}}
      >
        {price}
      </Text>
    </>
  );
};

const ImageSelection = ({
  id,
  imageLength,
  onChange,
}: {
  id: number;
  imageLength: number;
  onChange: React.Dispatch<React.SetStateAction<number>>;
}) => {
  return (
    <View
      style={{
        flexDirection: 'row',
        alignItems: 'center',
      }}
    >
      {[...Array(imageLength).keys()].map(index => (
        <Pressable
          key={index}
          style={{
            paddingHorizontal: 10,
            paddingVertical: 15,
          }}
          onPress={() => {
            onChange(index);
          }}
        >
          <Badge
            style={{
              backgroundColor: id == index ? '#000000' : '#cccccc',
            }}
            size={id == index ? 12 : 8}
          />
        </Pressable>
      ))}
    </View>
  );
};

export default FeaturedProduct;
