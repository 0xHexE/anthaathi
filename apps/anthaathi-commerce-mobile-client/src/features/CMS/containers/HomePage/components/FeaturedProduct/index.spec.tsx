import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';
import FeaturedProduct from './index';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';

describe('FeaturedProduct', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <FeaturedProduct
            productInfo={{
              name: 'test',
              price: 0,
              description: 'test',
              currency: 'USD',
              image: [
                'https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1',
              ],
            }}
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('featuredProduct')).toBeTruthy();
  });
});
