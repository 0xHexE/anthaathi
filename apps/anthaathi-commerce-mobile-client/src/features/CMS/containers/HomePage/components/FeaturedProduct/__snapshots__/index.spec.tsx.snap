// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedProduct should render the item 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginHorizontal": 12,
      "marginVertical": 10,
    }
  }
  testID="featuredProduct"
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "width": "50%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://burst.shopifycdn.com/photos/tea-cup-with-hot-peppers-and-yellow-tomatoes-on-red.jpg?width=240&format=pjpg&exif=1&iptc=1",
        }
      }
      style={
        Object {
          "borderRadius": 4,
          "height": 280,
          "width": "70%",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "paddingHorizontal": 10,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          collapsable={false}
          numberOfLines={1}
          style={
            Object {
              "alignSelf": "flex-end",
              "backgroundColor": "#000000",
              "borderRadius": 6,
              "color": "rgba(255, 255, 255, 1)",
              "fontSize": 6,
              "height": 12,
              "lineHeight": 12,
              "minWidth": 12,
              "opacity": 1,
              "overflow": "hidden",
              "paddingHorizontal": 3,
              "textAlign": "center",
              "textAlignVertical": "center",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "width": "50%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "marginBottom": 9,
          },
        ]
      }
    >
      test
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "500",
          },
        ]
      }
    >
      Price
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "fontSize": 16,
            "fontWeight": "500",
            "marginBottom": 9,
          },
        ]
      }
    >
      $0.00
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontSize": 22,
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 28,
          },
          Object {
            "textAlign": "left",
          },
          Object {
            "writingDirection": "ltr",
          },
          Object {
            "fontSize": 12,
            "fontWeight": "400",
            "marginBottom": 9,
          },
        ]
      }
    >
      test
    </Text>
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#313652",
              "borderRadius": 1,
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginBottom": 10,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 1,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#313652",
                      },
                    ],
                  ]
                }
              >
                Add to cart
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "#313652",
              "borderColor": "transparent",
              "borderRadius": 1,
              "borderStyle": "solid",
              "borderWidth": 0,
              "marginBottom": 10,
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 1,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 24,
                        "marginVertical": 10,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(255, 255, 255, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#ffffff",
                      },
                    ],
                  ]
                }
              >
                Buy it now
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
